include/eve/memory/soa_vector.hpp

```diff
@@ -0,0 +1,268 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/algo/zip.hpp>
+#include <eve/detail/kumi.hpp>
+#include <eve/function/read.hpp>
+#include <eve/function/write.hpp>
+#include <eve/memory/aligned_allocator.hpp>
+#include <eve/product_type.hpp>
+#include <vector>
+
+namespace eve
+{
+  //================================================================================================
+  //! @addtogroup memory
+  //! @{
+  //!   @brief SIMD-aware container for product types
+  //!
+  //!   **Required header:** `#include <eve/memory/soa_vector.hpp>`
+  //!
+  //!   soa_vector is a container similar to `std::vector` but optimized to store
+  //!   [product types](eve::product_type) using a Structure Of Array layout compatible with SIMD
+  //!   processing.
+  //!
+  //!   @tparam Type  Value type stored inside `eve::soa_vector`
+  //! @}
+  //================================================================================================
+  template<eve::product_type Type> struct soa_vector
+  {
+    private:
+    template<typename T> struct soa_storage
+    {
+      using type = std::vector<T, aligned_allocator<T,detail::common_cardinal<T>>>;
+    };
+
+    public:
+
+    //! Type of data stored in the container
+    using value_type = Type;
+
+    using storage_type = kumi::as_tuple_t<kumi::result::flatten_all_t<Type>, soa_storage>;
+
+    //==============================================================================================
+    //! @name Constructors
+    //! @{
+    //==============================================================================================
+
+    //! Constructs an empty container.
+    soa_vector() {}
+
+    //! Constructs the container with `n` default-inserted instances of `Type`. No copies are made.
+    soa_vector(std::size_t n) { resize(n); }
+
+    //! Constructs the container with `n` copies of elements with `value` value.
+    soa_vector(std::size_t n, value_type value) { resize(n,value); }
+
+    //! Constructs the container with the contents of the initializer list `l`.
+    soa_vector( std::initializer_list<Type> l )
+    {
+      resize( l.size() );
+      auto ptr = l.begin();
+      for(std::size_t i = 0;i < size(); ++i)
+      {
+        kumi::for_each( [&](auto m, auto& v) { v[i] = m; }, *ptr++, storage );
```

Comment:
Wait, what `operator*`? We removed all `operator*` and left only `read/write`

Reply:
This is * on the `initializer_list`